packages/dmn-editor/tests/e2e/drdArtifacts/deleteAssociationWaypoint.spec.ts

```diff
@@ -0,0 +1,135 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { expect } from "@playwright/test";
+import { test } from "../__fixtures__/base";
+import { DefaultNodeName, NodeType } from "../__fixtures__/nodes";
+import { TestAnnotations } from "@kie-tools/playwright-base/annotations";
+
+test.beforeEach(async ({ editor }) => {
+  await editor.open();
+});
+
+test.describe("Delete edge waypoint - Association", () => {
+  test.beforeEach(async ({ palette, nodes, browserName }) => {
+    test.skip(browserName === "webkit", "https://github.com/apache/incubator-kie-issues/issues/991");
+    test.info().annotations.push({
+      type: TestAnnotations.REGRESSION,
+      description: "https://github.com/apache/incubator-kie-issues/issues/991",
+    });
+
+    await palette.dragNewNode({ type: NodeType.INPUT_DATA, targetPosition: { x: 100, y: 100 } });
+    await nodes.dragNewConnectedNode({
+      from: DefaultNodeName.INPUT_DATA,
+      type: NodeType.TEXT_ANNOTATION,
+      targetPosition: { x: 100, y: 300 },
+    });
+  });
+
+  test("should delete the single Association edge waypoint", async ({ edges }) => {
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
+
+    await edges.deleteWaypoint({
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.TEXT_ANNOTATION,
+      waypointIndex: 1,
+    });
+
+    await expect(
+      await edges.getWaypoint({
+        from: DefaultNodeName.INPUT_DATA,
+        to: DefaultNodeName.TEXT_ANNOTATION,
+        waypointIndex: 1,
+      })
+    ).not.toBeAttached();
+  });
+
+  test("should delete the single Association edge waypoint to make it straight", async ({ diagram, edges }) => {
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
+    await edges.moveWaypoint({
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.TEXT_ANNOTATION,
+      waypointIndex: 1,
+      targetPosition: { x: 500, y: 300 },
+    });
+
+    await edges.deleteWaypoint({
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.TEXT_ANNOTATION,
+      waypointIndex: 1,
+    });
+    await expect(diagram.get()).toHaveScreenshot("delete-association-waypoint-straight-edge.png");
+  });
+
+  test("should delete all Association edge waypoints", async ({ nodes, edges }) => {
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
+    await nodes.move({ name: DefaultNodeName.TEXT_ANNOTATION, targetPosition: { x: 200, y: 500 } });
```

Comment:
please, use the move waypoint.

Reply:
here I think I can not incorporate your suggestion, currently the only way to have two or more waypoints is to:
- add first waypoint
- move one ending node, but the middle of the edge needs to be the same as middle between nodes, as adding waypoint simply click into the middle of the edge bounding box
- add next waypoint
- move ending node in the same manner as previously
- ...

Reply:
Ok, maybe I didn't paid attention to this detail on the other tests. But as I see now, the `{ x: 200, y: 500 }` is not an arbitrary value. I don't think this is a good approach as it works in a "mysterious" way. I mean, we don't have a comment explaining it, and we need to think that maybe we'll not remember this in a couple of months. We could add a very good comment for our future selfs on all tests, or we could change this so we can click on the middle of an edge segment whenever we want.

On the `path` element we have the `d` attribute, which is a string with the edge points. `M` as the starting point and `L` as the ending point. For an edge with multiple waypoints it will have as many `M` and `L` pairs, creating the edge segments. Our Diagram has an offset of `100,100`, which means if the point marks `100,100` it's actually `200,200`. Parsing the string, and knowing the two points `M` and `L` can tell exactly where is the middle point of the edge segment. For example, an edge with one waypoint can have the following `d` attribute `M 140,140 L 260,80 M 260,80 L 380,140`. If we want to click on the middle of the first edge segment: 

```
M 140, 140 (starting)
L 260, 80 (ending)

calculate the middle point
400, 240 (sum)
200, 120 (divide by 2)
300, 240 (add offset)
```

Clicking at `300,240` should click at the middle of the first edge segment if we don't have any other offset to apply. Now, this could be translated to a method on the edge fixture. `addWaypointBetweenConsecutiveWaypoints(args: { from: string, to: string, startingWaypoint: number  })`. I don't think `startingWaypoint` is a good name, but I couldn't think on a better one. The `startingWaypoint` would give us which pair we need to use from the `d` string.